### DISCUSSION TITLE
Interaction Deferring

### DISCUSSION THREAD

**Discussion by StoneMasons4106** [Pycord-Development/pycord#1892] [Q&A] [answered]
Hello all, I'm running into an issue where deferring a command that will take a while is not returning a response.

I'm attempting to use ctx.defer and ctx.followup.send and using this article for reference material:
https://github.com/Pycord-Development/pycord/issues/264

I'm attempting to find a specific user associated with the bot to find servers where patrons are to grant their servers extra commands, however the bot has grown in size and guild number so this presents a unique issue. 

Only I will have access to this command as you can see from the author id parameters, but as of now it's not returning a message. I'm aware this may not be the cleanest way to do this, any suggestions are welcome. My code so far is as follows:

```
@bot.slash_command(name='find-user', description='Finds a given user and returns their server info related to the bot.')
async def find_user(ctx, *, member):
    author_id = os.environ.get('AUTHOR_ID')
    message_author_id = str(ctx.author.id)
    if message_author_id == author_id:
        await ctx.defer(ephemeral=True)
        for guild in bot.guilds:
            for member in guild.members:
                if f'{member.name}#{member.discriminator}' == member:
                    await ctx.followup.send(f'{member} - {guild.id}', ephemeral=True)
                    break
                else:
                    continue
    else:
        await ctx.respond('You do not have access to this command.')

**Comment by JustaSqu1d** [comment, marked as answer]
`discord.User` has a [`mutual_guilds` attribute](https://docs.pycord.dev/en/stable/api/models.html#discord.User.mutual_guilds). You can use this instead.

I would like to add that this discussion category should not be used for help, rather, please join our [Discord server](https://pycord.dev/discord) for help instead. If you require further assistance please do the same.
